src/aggregate.d

```diff
@@ -270,35 +276,58 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             return 0;
         }
 
-        fields.setDim(0);
-
         for (size_t i = 0; i < members.dim; i++)
         {
             auto s = (*members)[i];
             if (s.apply(&func, cast(void*)this))
+            {
+                if (sizeok != SIZEOKnone)
+                    return true;
```

Comment:
There's some back and forthing going on, but it looks like this mostly relies on setting `sizeok` at the correctly.  It should always be `SIZEOKnone` unless the aggregate has been finalized (`SIZEOKdone`) or we tried determining it's contents but met with errors, likely recursive (`SIZEOKfwd`).

```
determineSize: => sizeok == SIZEOKnone
|-> determineFields => sizeok == SIZEOKnone  (fields.dim == 0, fields.setDim(0))
| |-> determineSize: => sizeok == SIZEOKnone  (fields.dim == 2)
| | |-> determineFields => sizeok == SIZEOKfwd (fields.dim == 2, fields.setDim(0))
| | |-> finalizeSize => sizeok == SIZEOKdone (fields.dim == 3, fields.setDim(0), setFieldOffset re-adds fields)
| |   |-> determineSize: => sizeok == SIZEOKdone  (fields.dim == 3)
| |-> determineSize: => sizeok == SIZEOKdone  (fields.dim == 3)
| |-> determineSize: => sizeok == SIZEOKdone  (fields.dim == 3)
| # Exits early at this particular line because sizeok != SIZEOKnone
# Does not call finalizeSize because sizeok == SIZEOKdone
```